hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Lists.java

```diff
@@ -232,4 +232,20 @@ private static int saturatedCast(long value) {
     return addAll(addTo, elementsToAdd.iterator());
   }
 
+  public static <T> List<List<T>> partition(List<T> originalList, int pageSize) {
+
+    Preconditions.checkArgument(originalList != null && originalList.size() > 0,
+            "Invalid original list");
+    Preconditions.checkArgument(pageSize > 0, "Page size should " +
+            "be greater than 0 for performing partition");
+
+    List<List<T>> result = new ArrayList<>();
```

Comment:
this always returns a new list even when the source is in range. seems like overkill

Reply:
subList() which is used below will return a view list only with the underlying storage of original list.